jans-client-api/server/src/main/java/io/jans/ca/server/filter/AuthorizationFilter.java

```diff
@@ -0,0 +1,82 @@
+ *
+ * Copyright (c) 2020, Janssen Project
+ */
+
+package io.jans.ca.server.filter;
+
+import io.jans.ca.server.security.service.AuthorizationService;
+import io.jans.ca.common.rest.ProtectedApi;
+import jakarta.annotation.Priority;
+import jakarta.inject.Inject;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.ws.rs.Priorities;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.container.ResourceInfo;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriInfo;
+import jakarta.ws.rs.ext.Provider;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+@Provider
+@ProtectedApi
+@Priority(Priorities.AUTHENTICATION)
+public class AuthorizationFilter implements ContainerRequestFilter {
+
+    private static final String AUTHENTICATION_SCHEME = "Bearer";
+    private static final String AUTHORIZATION_RP_ID = "AuthorizationRpId";
+
+    private static final Logger log = LoggerFactory.getLogger(AuthorizationFilter.class);
+
+    @Context
+    UriInfo info;
+
+    @Context
+    HttpServletRequest request;
+
+    @Context
+    private HttpHeaders httpHeaders;
+
+    @Inject
+    AuthorizationService authorizationService;
+
+    @SuppressWarnings({"all"})
+    public void filter(ContainerRequestContext context) {
+        log.info("=======================================================================");
+        log.info("====== context = " + context + " , info.getAbsolutePath() = " + info.getAbsolutePath()
+                + " , info.getRequestUri() = " + info.getRequestUri() + "\n\n");
+        log.info("====== info.getBaseUri()=" + info.getBaseUri() + " info.getPath()=" + info.getPath()
+                + " info.toString()=" + info.toString());
+        log.info("====== request.getContextPath()=" + request.getContextPath() + " request.getRequestURI()="
+                + request.getRequestURI() + " request.toString() " + request.toString());
+
+        log.info("======" + context.getMethod() + " " + info.getPath() + " FROM IP " + request.getRemoteAddr());
+        log.info("======PERFORMING AUTHORIZATION=========================================");
+        String authorizationHeader = context.getHeaderString(HttpHeaders.AUTHORIZATION);
+        String authorizationRpIdHeader = context.getHeaderString(AUTHORIZATION_RP_ID);
+
+        log.info("\n\n\n AuthorizationFilter::filter() - authorizationHeader = " + authorizationHeader + " , authorizationRpIdHeader = "
+                + authorizationRpIdHeader + " \n\n\n");
+        try {
+            authorizationService.processAuthorization(info.getPath(), context.getMethod(), request.getRemoteAddr(), authorizationHeader, authorizationRpIdHeader);
```

Comment:
In case `AuthorizationRpId` header is null, then it is read from body params:
https://github.com/JanssenProject/jans/blob/e44ed18da41f7621489cbeabb47491341de8a82c/jans-client-api/server/src/main/java/io/jans/ca/server/op/BaseOperation.java#L84

I tried to read the body content in filter, but then the body is not available for `io.jans.ca.server.op.BaseOperation#process` method.

`io.jans.ca.server.filter.AuthorizationFilter` process the token validation only when `AuthorizationRpId` header is not null.

One option could be to establish `AuthorizationRpId` header as mandatory.
Otherwise, we can look for a way to read body in filter, without losing it.

Reply:
Right, lets read `authorizationRpId` from header and move it out of body. In general everything was in body because oxd was designed as backchannel app, we should re-think it since now it's web app.

Reply:
Now, ´authorizationRpId´ is mandatory in header.

Reply:
It is ready for review